src/main/java/com/google/sps/data/Room.java

```diff
@@ -0,0 +1,49 @@
+package com.google.sps.data;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.HashMap;
+import javax.servlet.http.HttpServletRequest;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.Gson;
+
+public class Room {
+    private ArrayList<String> members;
+    private LinkedList<Message> messages;
+    private ArrayList<String> videoUrls;
+    private RoomState state;
+    public static final Gson PARSER = new Gson(); 
+
+    /**
+      * Room constructor
+      * @param obj a JsonObject representing the configurations the Room creator specified
+      * @param urls an arraylist of the video urls of the initial Room playlist
+      * @return a new Room object
+      */
+    public Room(JsonObject obj, ArrayList<String> urls){
+        this.messages = new LinkedList<Message>();
+        this.state = new RoomState();
+        this.members = this.PARSER.fromJson(obj.getAsJsonArray("members"),ArrayList.class);
```

Comment:
This looks strange to me. Why JsonObject has only member string list but no other values? Intuitively, it's more reasonable to have just mapping from jsonObject to Room class and room class to json string.